File: Operation_Map.Server/Models/LineItem.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;
using System.Collections.Generic;


namespace Operation_Map.Server.Models
{
    public class LineItem
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string? Id { get; set; }
        public string? LineItemName { get; set; }
        public List<string>? Subcontractors { get; set; }
        public int? NumberBids { get; set; }
        public bool? SelectedBid { get; set; }
        public string? Notes { get; set; }
        public double? CompletionPercentage { get; set; }
        public List<string>? PaidInvoices { get; set; }
    }

}

File: app/shared/data/project/line-item.ts
import { Subcontractor } from './subcontractor';

export interface LineItem {
  id?: string;
  lineItemName?: string;
  numberBids?: number;
  budget?: number;
  subcontractors?: Subcontractor[];
  subcontractorCost?: number;
  notes?: string;
  bidSelected?: boolean;
  invoices?: [];
}
